app/main.py

```diff
@@ -1,2 +1,224 @@
+        self.size -= 1
+
+    def __iter__(self) -> Iterator[tuple[Hashable, Any]]:
+        for node in self.hash_table:
+            if node is not None:
+                yield node.key, node.value
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __str__(self) -> str:
+        items = [
+            f"{node.key}: {node.value}"
+            for node in self.hash_table
+            if node is not None
+        ]
+        return "{" + ", ".join(items) + "}"
+
+    def __eq__(self, other: Dictionary) -> bool:
+        """
+        Compare key, value pairs between two Dictionaries.
+        """
+        if not isinstance(other, Dictionary):
+            return False
+
+        if self.size != other.size:
+            return False
+
+        for node in self.hash_table:
+            if node is not None:
+                if other.__getitem__(node.key) != node.value:
+                    return False
+
+        return True
+
+    def get(self, key: Hashable, default: Optional[Any] = None) -> Any:
+        """
+        Get the value for the given key.
+        If the key does not exist, return the default value.
+
+        Default value is optional.
+        If it was not provided, method will raise KeyError.
+        """
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def pop(self, key: Hashable, default: Optional[Any] = object()) -> Any:
+        try:
+            value = self[key]
+            self.__delitem__(key)
+            return value
+        except KeyError:
+            if isinstance(default, object):
+                return default
+            else:
+                raise KeyError(
+                    f"Cannot delete key: {key} and "
+                    "no default value is provided"
+                )
+
+    def _update_from_dict(self, data: Dictionary) -> None:
+        """
+        Updates the dictionary with a Dictionary object.
+        """
+        for node in data.hash_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def _update_from_builtin_dict(self, data: dict) -> None:
+        """
+        Updates the dictionary with key-value pairs from builtin dictionary.
+        """
+        for key, value in data.items():
+            self.__setitem__(key, value)
+
+    def update(
+            self,
+            data: Dictionary | list[dict] | dict
+    ) -> None:
+        """
+        Updates Dictionary with various types of data.
+        """
+        if isinstance(data, Dictionary):
+            self._update_from_dict(data)
+        elif isinstance(data, dict):
+            self._update_from_builtin_dict(data)
+        elif isinstance(data, list):
+            for item in data:
+                if isinstance(item, dict):
+                    self._update_from_builtin_dict(item)
+                else:
+                    raise TypeError(f"Unsupported type {type(item)}")
```

Comment:
No need to raise those errors - if wrong types are being passed, python will trigger exceptions by itself. and we don't need to catch them here - they can be caught when we work with the object already